### DISCUSSION TITLE
UPDATED: Lovelace Card and Package For Playing Media Library Items from Home Assistant UI

### DISCUSSION THREAD

**Discussion by jazzyisj** [music-assistant/support#2990] [Show and tell]
**Updated - 2025-5-22**

- **Custom card requirements removed from lovelace card**
- **Filters now working correctly**

I have created a package with all the of the entities necessary to select, save, and play your **favourite** Music Assistant media items from a card in the Home Assistant UI, and another file containing the YAML for the associated lovelace card.  The only item to change to utilize this package in your HA config is to change the media player entity_id to any one of your Music Assistant media player entity_id's where indicated with the #TODO tag and add the lovelace card to your ui.  The `music_assistant.get_library` action call parameters can be changed to suit your requirements.

![mass_play_media_card](https://github.com/user-attachments/assets/4fd1ac71-3ad0-4f8b-811c-d0b440a16c73)

[Play MASS Media - Package](https://gist.github.com/jazzyisj/c53c92779785bb5d9952adf3a3a28107)
[Play MASS Media - Lovelace Card](https://gist.github.com/jazzyisj/f90bf1c1c3205d5d2d791f196fc144b1)

In my own config I use multiple versions of these entities and lovelace card to create various "media presets" that can be recalled and played automatically in response to selected events (morning, sleep, arriving home etc.) or at the click of a button.  If you're interested in how to accomplish this, just ask - I am happy to share.

**Comment by raevenlock** [comment]
Love this! Can't get it to work though, haha
Probably something I have missed/doing wrong. I can't get it to find any media_players in the drop down list.
I have tried to add my own of course, and tried to add some radio channels and albums and so on, but can't get it to work.

Right now 
**Media Type** is set to Radio
**MASS Radio** is set to my own Asterix Radio
**Media Player** ... _the list is empty_
Entity not available: input_number.media_volume_play

Can't figure it out.

(Music Assistant is running fine)

**Comment by jazzyisj** [comment]
OK, input_number.media_volume_play is easy to fix.  It should be input_number.mass_volume.  Copy and paste error on my part, I've updated the gist.  My apologies, it works on my system because I do have an input_number.media_volume_play entity so I didn't catch the error.

As for the media player dropdown list, did you reload automations, templates, and groups or restart Home Assistant?  If you have and you still don't have any media players in the dropdown can you copy this and paste it into the Developer Tools Template Tool and tell me what the result is?
```

          {{ states.media_player
              | selectattr('attributes.mass_player_id', 'defined')
              | map(attribute='entity_id') | list
              | select('has_value') | list }}
```

<img width="839" alt="image" src="https://github.com/user-attachments/assets/3dd94cb4-8278-4364-ac19-10c6be709dd9">

**Reply by raevenlock** [reply]
Hey and thnx for the reply! 
The input_number.media_volume_play issue, was as you said a pretty easy fix...  👍 

BUT, for my other problem.. I am just bad and I should feel bad! haha 
I had forgotten that I had changed HA to be in UI mode (or whatever its called)  a while back. So I can't just copy and paste, I need to actually create a new automation in the UI and paste your code in there instead and create a templete in the UI and so on.

So a bit more work on my part, I guess :)
I will be back with results.. probably during the weekend

Btw, your code for the "states.media_player" thing, that I should paste in the developer tools, is working fine! It listed all my media_players as your screenshot.

**Reply by jazzyisj** [reply]
If you enable packages in your configuration.yaml you can just copy the package file to that directory (usually /config/packages).  You won't have to add the templates, automation or group in the UI.

You will need to paste the yaml for the lovelace card wherever you want it on your dashboard like you would any other card.   If you are using the UI for your dashboard config create a new entities card and click show editor at the bottom.  Copy and paste the code there.

https://www.home-assistant.io/docs/configuration/packages/

<img width="831" alt="image" src="https://github.com/user-attachments/assets/1dec93bb-6597-445a-b137-2e64238728b9">

**Reply by raevenlock** [reply]
> If you enable packages in your configuration.yaml you can just copy the package file to that directory (usually /config/packages). You won't have to add the templates, automation or group in the UI.
> 
> You will need to paste the yaml for the lovelace card wherever you want it on your dashboard like you would any other card. If you are using the UI for your dashboard config create a new entities card and click show editor at the bottom. Copy and paste the code there.
> 
> https://www.home-assistant.io/docs/configuration/packages/
> 
> <img alt="image" width="831" src="https://private-user-images.githubusercontent.com/44064559/379907231-1dec93bb-6597-445a-b137-2e64238728b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk4Mzc3NzgsIm5iZiI6MTcyOTgzNzQ3OCwicGF0aCI6Ii80NDA2NDU1OS8zNzk5MDcyMzEtMWRlYzkzYmItNjU5Ny00NDVhLWIxMzctMmU2NDIzODcyOGI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI1VDA2MjQzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI4ZjZhZGZlOWMzNTM0YjcwNmNiNGQ2MGQ1NmViMjllZTE2YWFlMTg5MGMwNzU0ZmUzMWZlZGUyZWIwMTZiYTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.QG2mNDD9okjAYxe2_KlU6X6LX79m1MO8BoFYhR2sOFg">

Yeah, I will do it like this instead. Probably smarter.
Thanks for your help so far!

**Reply by tomadan1968** [reply]
> @tomadan1968 why are you writing in my post? Please start your own thread. Nobody likes a thread-stealer

Sorry, i corrected it, thank you!

**Reply by raevenlock** [reply]
> If you enable packages in your configuration.yaml you can just copy the package file to that directory (usually /config/packages). You won't have to add the templates, automation or group in the UI.
> 
> You will need to paste the yaml for the lovelace card wherever you want it on your dashboard like you would any other card. If you are using the UI for your dashboard config create a new entities card and click show editor at the bottom. Copy and paste the code there.
> 
> https://www.home-assistant.io/docs/configuration/packages/
> 
> <img alt="image" width="831" src="https://private-user-images.githubusercontent.com/44064559/379907231-1dec93bb-6597-445a-b137-2e64238728b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk4Nzk4NTQsIm5iZiI6MTcyOTg3OTU1NCwicGF0aCI6Ii80NDA2NDU1OS8zNzk5MDcyMzEtMWRlYzkzYmItNjU5Ny00NDVhLWIxMzctMmU2NDIzODcyOGI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI1VDE4MDU1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI5NmUzN2ZmMWI1MjU0OWVjY2ExYmRjYWY2MGU3OTA3OTRkNzMwOWM0YzkyNTMyMDc2ZDAyYjMxZGMxYjIzYTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.KsIpPTAIBa4I3KPP98n5x5OT4fvNXE2F3smv7l-_5JQ">

FYI got it to work. Thanks again and great work! Apriciate it

**Comment by tomadan1968** [comment]
I still have two problems, one the volume does not increase or decrease when I pull the slider and the second every time I change the playlist, a radio station or any media the volume of the player I am playing comes to zero and I always have to go to the respective player and increase the volume because nothing is heard, please help me to solve these 2 problems if possible, thank you!

**Reply by jazzyisj** [reply]
This card only starts the media play.  After the media has started you control everything with the media player.  You _could_ write automations to control the media player with this card's controls, but that is beyond the scope of what we're doing here.

**Reply by tomadan1968** [reply]
Ok, thank you !

**Comment by potelux** [comment]
I recently detailed a solution that could make sharing this sort of card much easier.  That comment can be found [here.](https://github.com/orgs/music-assistant/discussions/2601#discussioncomment-11326049)

Additionally, I made a [feature request on Home Assistant](https://community.home-assistant.io/t/dashboard-card-blueprints/709559) to make this all possible.  

If you think this feature would be helpful in sharing cards like this with the community, let me know.  I am looking for a project to do over Christmas.

**Comment by skavan** [comment]
This is fantastic and got me thinking.

1. I expanded your mass_play_meda.yaml card with a new section:
```
- trigger:
    - trigger: homeassistant
      event: start

    - trigger: event
      event_type: event_template_reloaded

    - trigger: state
      entity_id:
        - input_text.mass_playlist
        - button.mass_update_media
      to:
    
    - trigger: state
      entity_id:
        - input_text.mass_playlist_search
    action:
      - action: mass.get_library
        data:
          favorite: true
          limit: 500
          media_type: playlist
          order_by: play_count_desc
          search: |
            {{ states('input_text.mass_playlist_search') or null }}
        response_variable: mass_playlists

      - variables:
          playlists: |
            {% set playlist_list = namespace(info=[]) %}
            {% for playlist in mass_playlists['playlists'] %}
              {% set provider_domain = playlist.provider_mappings[0].provider_domain if playlist.provider_mappings else 'unknown' %}
              {% set provider_url = playlist.provider_mappings[0].url if playlist.provider_mappings else '' %}
              {% set image_path = playlist.metadata.images[0].path if playlist.metadata.images else 'unknown' %}
              {% set playlist_list.info = playlist_list.info + [{
                'item_id': playlist.item_id,
                'provider': playlist.provider,
                'name': playlist.name,
                'uri': playlist.uri,
                'media_type': playlist.media_type,
                'provider_domain': provider_domain,
                'provider_url': provider_url,
                'image_path': image_path
              }] %}
            {% endfor %}
            {{ playlist_list.info }}
```
2. I also created a new input: `input_text.mass_playlist_search`
3. And then below is the yaml mode view. It dynamically creates playlist buttons. You can apply a search and when you click on the buttons...its launches the playlist.
4. A bit of hard coding that needs to be fixed. But between us all, we should be able to create a little library of MA popups in HA. Screenshot at the end of this post.

```
button_card_templates:
  base:
    layout: icon_name
    styles:
      card:
        - height: 42px
      grid:
        - grid-template-areas: '"i n "'
        - grid-template-columns: 40px auto
        - grid-template-rows: 1fr
      img_cell:
        - justify-content: start
        - align-items: start
        - height: 44px
        - margin-top: -11px
      icon:
        - width: 30px
        - height: 30px
        - justify-self: start
        - padding: 0px 0px
        - margin-left: 8px
        - border-radius: 12px
      name:
        - padding: 4px 0px 0px 8px
        - max-width: 92%
        - justify-self: start
        - align-self: flex-start
        - font-size: 13.5px
        - font-weight: 400
        - margin-top: -2px

views:
- title: Music Assistant Playground
  type: custom:grid-layout
  layout:
    grid-template-columns: 100%
  cards:
    - type: entities
      entities:
        - input_text.mass_playlist_search
    - type: custom:config-template-card
      variables:
        playlist_options: |
          () => {
            const options = states['select.mass_playlist']?.attributes?.options || [];
            return options;
          }
        playlists: |
          () => {
            const playlists = states['sensor.mass_playlists']?.attributes?.playlists || [];
            return playlists;
          }
        alphabet: |
          () => {
            return Array.from({length: 26}, (_, i) => String.fromCharCode(97 + i));
          }
        createbuttons: |
          (playlists) => {
            if (playlists.length === 0) {
              return [{
                type: "custom:button-card",
                name: "No Playlists Available",
                template: ["base"],
                entity: "sensor.ui_remote",
                icon: "mdi:alert",
              }];
            }
            return playlists.map(playlist => {
              // const picture = playlist.image_path.startsWith('/') ? '/local/generic_playlist.png' : playlist.image_path;
              const picture = (!playlist.image_path.startsWith('http')) ? '/local/generic_playlist.png' : playlist.image_path;
              return {
                type: "custom:button-card",
                name: playlist.name,
                template: ["base"],
                entity_picture: picture,
                show_entity_picture: true,
                tap_action: {
                  "action": "perform-action",
                  "perform_action": "mass.play_media",
                  "data": {
                    "media_id": playlist.uri,
                  },
                  "target": {
                    "entity_id": "media_player.ma_connect_basement"
                  }
                }

              };
            });
          }
      entities:
        - sensor.ui_remote
        - sensor.mass_playlists
      card:
        type: custom:auto-entities
        entities: "${createbuttons(playlists())}"
        card: 
          type: custom:layout-card
          layout_type: custom:grid-layout
          layout:
            mediaquery:
              '(min-width: 900px)':
                grid-template-columns: repeat(5, 1fr)
              '(min-width: 420px) and (max-width: 899px)':
                grid-template-columns: repeat(2, 1fr)
              '(min-width: 300px)':
                grid-template-columns: repeat(auto-fill, minmax(150px, 1fr))
            grid-template-rows: auto
    - type: custom:button-card
      template: base
      name: > 
        [[[ return states['input_text.mass_playlist_search'].state ]]]
```

<img width="500" alt="image" src="https://github.com/user-attachments/assets/27d692f4-0a03-4781-9c5f-cf8d29df491d">

<img width="499" alt="image" src="https://github.com/user-attachments/assets/036d7a60-ece1-4587-b78e-66c25b6c05d1">

**Reply by tomadan1968** [reply]
Hi, can you show us the complete code for your mass_play_media.yaml? Thank you!

**Comment by larsalthof** [comment]
I am trying this, and 3 entities seem to be missing. Did I misspell something somewhere?
<img width="577" alt="image" src="https://github.com/user-attachments/assets/756e1182-ddb9-4df8-ab53-012c9e650708">

**Reply by jazzyisj** [reply]
You used the package as is?  Have you reloaded all components or restarted HA?

**Reply by larsalthof** [reply]
> You used the package as is? Have you reloaded all components or restarted HA?

I used it as-is except for updating the media players. I did restart HA as well.

**Reply by larsalthof** [reply]
> > You used the package as is? Have you reloaded all components or restarted HA?
> 
> I used it as-is except for updating the media players. I did restart HA as well.

I changed the card to reference these instead:
select.mass_media_player
input_boolean.mass_repeat
input_select.mass_enqueue_mode

So far, so good

**Reply by jazzyisj** [reply]
Oh shoot, I uploaded an out of date card again.  Second time I've done this now.  Ugh.  I don't catch it because those entities DO exist in my config...  

I just updated both the package and the card gists to fix this issue and a couple of other tweaks.  If you have time and are willing do you updating your package and lovelace card to the new version and testing it for me?  Trying to avoid anyone else experiencing this frustration.  My apologies btw.

**Reply by larsalthof** [reply]
Looks like it is working. I am a little embarrassed that it took me that long to identify that the select is generated dynamically.

Thank you

**Comment by tomadan1968** [comment]
Hello, is it still working? I can't find any playlists, radio stations, artists...nothing...after the home assistant updates...

**Reply by OzGav** [reply]
Is what still working? If you mean Music Assistant then all is well.

**Reply by tomadan1968** [reply]
Was it a joke? normally I mean Lovelace Card and Package For Playing Media Library Items from Home Assistant UI...

**Reply by OzGav** [reply]
Ok sorry. No idea

**Reply by jazzyisj** [reply]
@tomadan1968 - No this wasn't working.  This old code was from before Music Assistant was made a core integration.  The service calls have changed.  I have updated both the package and the lovelace card to work with the core integration.  Download the new versions and you should be good to go.

@OzGav - While working on this I noticed get_library haven't been updated to include podcasts and audio books, at least in the ui.  I don't have either so I haven't tested it.

**Reply by tomadan1968** [reply]
Entity get_library missing in home assistant..

**Reply by jazzyisj** [reply]
@tomadan1968   You're not giving me much to work with here with to help you.

- How did you install this?  Package method? (recommended)
- Have you restarted Home Assistant?
- Where are you seeing this error?  In your system log, in the ui?

**Comment by warthog10** [comment]
Need a little assistance here. I loaded the card using the package and all seems like it should work. The Media Type dropdown works and changes to artist, album, track, etc.; however, the second dropdown never returns any data, no matter what media type I choose. Not sure where I went wrong here.


![Screenshot 2025-03-06 170901](https://github.com/user-attachments/assets/0a39b262-e0b3-4bb2-9413-4bd053f3193d)

**Reply by jazzyisj** [reply]
Have you "favourited" any artists, playlists etc in Music Assistant?  Only favourite items get returned unless you change the parameters in the service call.

**Comment by GerritKuilder** [comment]
I am getting the following errors:
```
Configuration warnings
Setup of package 'input_boolean' failed: Integration 'mass_shuffle' not found.
Setup of package 'input_boolean' failed: Integration 'mass_radio' not found.
Setup of package 'input_select' failed: Integration 'mass_enqueue_mode' not found.
Setup of package 'input_select' failed: Integration 'mass_media_type' not found.
Setup of package 'input_select' failed: Integration 'mass_repeat_mode' not found.
Setup of package 'input_text' failed: Integration 'mass_media_player' not found.
Setup of package 'input_text' failed: Integration 'mass_artist' not found.
Setup of package 'input_text' failed: Integration 'mass_album' not found.
Setup of package 'input_text' failed: Integration 'mass_track' not found.
Setup of package 'input_text' failed: Integration 'mass_playlist' not found.
Setup of package 'input_text' failed: Integration 'mass_radio' not found.
Setup of package 'input_text' failed: Integration 'mass_artist_filter' not found.
Setup of package 'input_text' failed: Integration 'mass_album_filter' not found.
Setup of package 'input_number' failed: Integration 'mass_volume' not found.
Setup of package 'template' failed: Invalid package definition 'template': expected a dictionary. Package will not be initialized
Setup of package 'automation' failed: Invalid package definition 'automation': expected a dictionary. Package will not be initialized
```
Am I missing something? I have added the packages directory as
homeassistant:
  packages: !include_dir_named packages

Kind Regards,

Gerrit

**Reply by jazzyisj** [reply]
Confirming you added the package directory like the following in your configuration.yaml file (indentation matters) and that directory is actually `/config/packages`.

```
homeassistant:
  packages: !include_dir_named packages
```

How did you download the package file from GitHub and put it in your /config/packages directory?  Did you make any changes to the file other than changing the media player entity id where indicated?  For some reason it is looking at the integration (eg input_boolean) as the package name and the elements of the integration (eg. mass_shuffle as an integration) which I'm pretty sure means the indentation got messed up somewhere.

Remove whatever you have in the /config/packages directory right now.  Run a configuration check to verify there are no errors in your config without the package file.

Download a new copy of **the zipped package file** from Github and extract the mass_play_media.yaml file directly from the zip and place the file **with no changes** in the /config/packages directory.  

Now run a configuration check.  What happens?

**Reply by GerritKuilder** [reply]
:q!So I now placed 
homeassistant:
  packages: !include_dir_named packages

Completely at the top, I don't see errors amymore, 

but cannot select anything no albums/artists etc show up, cannot select the player (although it is the specific player for music assistant (Living Room media_player.living_room_2 Music Assistant) and I can select music from music master when I select that player...
I am running HA from a docker and have a separate docker for music assistant.

**Reply by GerritKuilder** [reply]
Update Mass Media Players uses an unknown action
Error ⸱ Reported by Automation
The automation "Update Mass Media Players" (automation.update_mass_media_players) has an unknown action: group.set.

This error prevents the automation from running correctly. Maybe this action is no longer available, or perhaps a typo caused it.

**Reply by jazzyisj** [reply]
Well that doesn't make sense, `group.set` is a core service.  Did you make any changes to the automation? 
Can you copy and paste the entire Update Mass Media Players automation here? (format it as code please!)

**Reply by GerritKuilder** [reply]
No other changes than changing the media player....
```
      - action: group.set
        data:
          object_id: mass_players
          entities: >
            {{ states.media_player
                | selectattr('attributes.mass_player_type', 'defined')
                | map(attribute='entity_id') | list
                | select('has_value') | sort }}
```

**Reply by jazzyisj** [reply]
Huh.  Running out of ideas here then.  I've verified the package works without changes in my config.  Can't imagine would would cause group.set to not be available.  I'm assuming you're using a recent version of HA?  Have you actually restarted HA? Are you using the beta version of music assistant addon?  Also, the automation error should not prevent the mass selects from populating.  Have you enabled the web service in music assistant?

<img width="290" alt="image" src="https://github.com/user-attachments/assets/321b20fa-d3e5-4082-bd1f-10b13c1b98aa" />

**Reply by GerritKuilder** [reply]
All latest versions, restarted a few times.
Stopped the containers.

I have a media player for MA where I can browse to music etc.  and play from there.

[Edit: could..., I added the HA integration to music assistant and things are pretty screwed up....]

I will need to start from fresh, at some stage]

**Reply by GerritKuilder** [reply]
So was able to update everything:
on MA enabled  the home assistant plugin (not done before)
Used only the Sonos s1 provider.

IN HA I have 2 players for MA 1 kitchen and one living room, plus the normal Sonos players.

Still the same error, bummer....

**Reply by jazzyisj** [reply]
Gotcha.  Well I'm sorry but I'm all out of ideas for you at this point.   I know this works well - I use several version of it myself and I installed the version on the gist to verify it still works on my system without error.  Something's gotta be buggered up somewhere.  I really don't know what would cause a core service to be unavailable.

**Reply by GerritKuilder** [reply]
Yeah, no problem. Thanks for your efforts and help.
Appreciated...

**Comment by JDogg016** [comment]
Have you given thought to a search capability within the lovelace card.  For example, if you want to play a pop music playlist could you type that information somewhere.

**Reply by jazzyisj** [reply]
(https://github.com/orgs/music-assistant/discussions/2990#discussioncomment-11341549)

**Comment by JDogg016** [comment]
Bravo

**Comment by SmileyRG** [comment]
@jazzyisj Thank you for the package & card.  I like the simple layout. Relatively new to HA and very new to MA but I think I have it all setup correctly but the Play Media (press) button does not seem to do anything.  I can go to MA and I can select a player (using chromecast players) and an artist or album or track and it plays as anticipated. However, from the HA card the dropdowns work correctly but nothing happens when I press 'play media'. For example, I have a chromecast player in my home office and in MA it is named Office_Cast. I used the entity from Office_Cast which is media_player.office for #TODO.  As noted, I do see anticipated values in the drop downs.  In the beginning I think I had it confused as I had the HA integration for Google Cast and then the MA player Chromecast plus I had the MA player for HA so I had 2 of everything.  At this point in time, I removed the HA Google Cast integration and have the MA Chromecast Player Provider.  As I was writing this thought of a possible conflict, I am using Plex Server and I have Plex Server HA integration and I have MA Plex Server Provider. Could this be a conflict and should I only have Plex Server MA Provider and delete the HA integration?  Otherwise, any thoughts / suggestions would be greatly appreciated.  It feels like I am 1 yard from a goal line but cannot seem to find the missing link.  Thanks in advance for any assistance you can provide.

**Reply by jazzyisj** [reply]
There would be no conflict leaving the cast integration in HA and also in MA.  You end up with 2 separate (but linked) media player entities.  eg.  `media_player.office`  and `media_player.office_2`.  I have both enabled in my system.

So you are getting your playlists, artists and all that fun stuff in your drop downs?  All the expected media players are in the media player dropdown?  Does anything happen?  Is it playing but the volume is 0?  Do you see any errors in your log?

**Comment by SmileyRG** [comment]
Thank you for the super fast response!

I am not sure if it was…

That I kicked it enough (haha)

Removed Plex Media Server integration from HA

Or, I gave it time to process after the restart (which I did many restarts today as I tried different scenarios)

I threw in the towel and did a HA restart and then went to cook dinner on the grill and while I was waiting for the grill to warm up, I gave it a try on my mobile and it worked!

BTW..  what I saw when I had the cast/chromecast in both HA and MA, I had to use the HA media player entity to get the dropdowns to work (#TODO) but no playback on any media player selected. That is why I removed the Google Cast from HA and only have Chromecast in MA for players.  After I did he changes I do not believe the group created for the players was updating so I just added ‘1’ to the end of the name in the code and then the group looked correct.  I used developer tools > actions to delete the old group since it would not delete thru the UI. I was not looking at the log (dumb me) but before I stopped for dinner I checked the log.  I did see a message that the script (py) that was looking for a track (track was selected) and ‘track name’ could not be found.  At this time the dropdowns look great (I changed the limits) and I have a list of cast players that I was planning on.

Again, Thank you for the code and the response!

I will continue to play!

Sorry for reaching out but like I said, I had thrown in the towel and was not sure what to try next.

**Reply by jazzyisj** [reply]
No worries, glad you got it working.

**Comment by SmileyRG** [comment]
Spoke too soon…

 

Here is the log file error..

 

Logger: homeassistant.helpers.script.mass_play_media_press
Source: helpers/script.py:2067
First occurred: 7:01:18 PM (2 occurrences)
Last logged: 7:02:03 PM

*	MASS Play Media press: Error executing script. Error for call_service at pos 4: Could not resolve ['Kid Charlemagne'] to playable media item
*	MASS Play Media press: Already running

 

I believe the 2 times is because I tried 2 different players.

 

This is the same message I saw before but for a different selected track.

 

So..  maybe a timeout that is causing the script to stop and not find the file / track?

 

I just used pc browser (chrome) and found the track in MA and it played on the correct player.

 

Note: running HA Green and using win11 PC / Chrome browser with 2 android mobile phones.

**Comment by jazzyisj** [comment]
It has nothing to do with the media player.  Read the error.

`Could not resolve ['Kid Charlemagne'] to playable media item`

It looks like everything is working ok with this package but MA isn't able to resolve the media item.

Have you tried different playlists, artists, tracks?

Put this into dev tools - template and let me know what the result is.

`{{ states('select.mass_' ~ states('input_select.mass_media_type') | lower) }}`

**Comment by SmileyRG** [comment]
I think I found the issue. If I select music at the beginning of the 'type'
it works.  Example, selected album AC/DC and it played.  Also tried the one
Playlist with one song and it worked.  I will need to build some Playlist
and use them vs artist/album/track.

**Reply by jazzyisj** [reply]
The artist album track should work!  If I get a chance later I'll reinstall the package on my (I use a highly customized version) and see if there's an issue somewhere, but a quick look at the code it looks ok.  Keep in mind it's designed to only work with items you have "favorited" in your MA library.  Also, make sure to click the "Update Media" button if you make any changes to your MA library.  (It will eventually update on its on though).   Also try radio mode - ie, select an artist and turn on radio mode.   Also make sure that you have enqueue mode set correctly.  If you want something to play right away it has to be Play or Replace otherwise you're just adding your selection to the queue.

**Comment by SmileyRG** [comment]
Thanks again.
Please do not load as I am positive it is timeouts.  I set favorites to
false and have 680 albums.  I will set favorites to true and mark favorites
and reset the limits. I think it is still usable but I need to adjust
expectations.  I really do appreciate your response and everything you have
done.

**Reply by jazzyisj** [reply]
Yep, inputs aren't really designed to handle thousands of options.  That's a lot of data to keep in the UI!

I run I think 10 different copies of this that represent different stored "Media Presets" that I can easily set from the UI when I want to change things up.  I have no issues, but I am also selective with my favorites and keep the limits set quite low (100 I think?)

Anyway, no trouble at all, hopefully the troubleshooting helps other people out too. Cheers.

**Comment by larsalthof** [comment]
Hi, I just updated package and card. I am not getting anything in the artist or album filters although I do see both albums and tracks the dropdown. Is there something I need to configure?

**Reply by larsalthof** [reply]
Updating the "select" names here did improve it, but now it just adds the one that I have already selected. Not sure I understand the intention.
<img width="589" alt="image" src="https://github.com/user-attachments/assets/a0175a1a-1c81-4891-ad41-52559d3a543f" />

**Reply by jazzyisj** [reply]
Sorry for delayed reply, I wanted to fix it first.  Apparently I was in a hurry and forgot to finish up the filters before I uploaded it.  Anyway, I've updated the gist.   Both the package and the card.  Download the new version.  It should be working now.

The track filters help you find a specific track.  Say you had several covers of the song "Wonderwall" in your library but you only want to play the version by Oasis.  "Oasis" should be in the artist filter list if you select the track Wonderwall.  If Oasis had more than one album with Wonderwall on it you can then filter it by the desired album.  You can also skip the artist filter and go directly to the album filter. 

You can use an artist filter you've selected an album with a title shared by several other albums (eg "Greatest Hits") .  You can filter your list by artist to get the desired album.

**Reply by larsalthof** [reply]
Thank you, and thanks for explaining the intent with the filters. In the meantime, I added audiobooks. How to best share?

**Reply by jazzyisj** [reply]
I can audiobooks and podcasts to the gists when I get a minute.  Both the package and the card will need to be updated.

**Comment by Yolnon** [comment]
Hello, after selecting the artist, album, radio... and pressing Play Media I get this error message:
Could not perform action button/press. string value is None for dictionary value @ data['artist']

Has anyone had this same problem?

**Reply by jazzyisj** [reply]
@Yolnon - They must have changed something in the service call.  It used to accept null for the artist filter. I've updated the gist with a fix.  Let me know if it works for you.

**Reply by Yolnon** [reply]
Working properly, Thank you

**Comment by robertw** [comment]
"In my own config I use multiple versions of these entities and lovelace card to create various "media presets" that can be recalled and played automatically in response to selected events (morning, sleep, arriving home etc.) or at the click of a button. If you're interested in how to accomplish this, just ask - I am happy to share."

if you have where and if not could you thank you cant see it in the thread thank you

**Comment by Pjarbit** [comment]
First, I added a stop button that stops the playing media.  If anyone wants the yaml for it please let me know.  Second, I have music assistant on many different panels.  Music assistant puts "this panel" as an available media player, but I dont know how to access it with this card/integration anyone have any ideas.  Overall this card does exactly what I want it to do and it's really great.  Thanks for all your hard work.  I am appreciative.
